### ISSUE TITLE
API key invalid and error thrown

### ISSUE THREAD

**Issue by stanislavkozlovski** [brianpetro/obsidian-smart-connections#748]
When I add the API key I get an error. It's freshly created from https://platform.openai.com/api-keys
<img width="188" alt="image" src="https://github.com/user-attachments/assets/210a03e5-d974-4e5e-b2ff-8e46037fc610">

When I try to chat with it, I get an error:
<img width="1176" alt="image" src="https://github.com/user-attachments/assets/933f6a86-a9b2-4ad8-8ec7-ad4083637152">

The full response is:
```
"{
    "error": {
        "message": "Incorrect API key provided: undefined. You can find your API key at https://platform.openai.com/account/api-keys.",
        "type": "invalid_request_error",
        "param": null,
        "code": "invalid_api_key"
    }
}
"
```

**Comment by stanislavkozlovski** [comment]
<img width="534" alt="image" src="https://github.com/user-attachments/assets/48bb7d35-3ad9-421b-af5f-4bb0692d5395">

It seems like it's not adding it at all? But I do see it in the settings

<img width="717" alt="image" src="https://github.com/user-attachments/assets/cc3b43ed-ebd3-4f9d-8576-08bfca8baba3">

**Comment by brianpetro** [comment]
@stanislavkozlovski The API key setting can be found in the Smart Chat window, see this screenshot
![Screenshot 2024-08-29 at 7 27 02 PM](https://github.com/user-attachments/assets/02c062ca-208d-469f-827c-78ce4947235a)

**Comment by carsten0202** [comment]
I have a related problem, except I cannot enter my API Key, because nothing happens when I click the settings button indicated by the green arrow.

**Comment by brianpetro** [comment]
@carsten0202 does an error appear in the console logs when you click the button? 🌴

**Comment by cheloreyna** [comment]
I am getting the same response 
![image](https://github.com/user-attachments/assets/d6d2b2e2-66bc-4cf2-b4b2-a30b1f8dddf2)

I'm using Gemini API 
![image](https://github.com/user-attachments/assets/106af2fb-5ed9-40d8-8a1e-dabfbe4bb5f1)

I would paste logs but i'm not sure how to access them

**Comment by carsten0202** [comment]
Yes. I get both an error when starting Obsidian and when clicking the button. It complains about port 37042 being in use. I would try changing it, but I can't find where to do that?

I've pulled a screenshot. The topmost errors are from startup, the one at the bottom is from when I click the settings button.

![Screenshot 2024-09-03 141343](https://github.com/user-attachments/assets/02d6145c-0dbc-45c8-8bd0-1d325f19fcca)

**Comment by AaronArt** [comment]
In my case I even have the paid version of ChatGPT. There is an alert window that says the API is valid, but when I try to chat, it says API error and when I look to the console log , it says, invalid API. So???
